barretenberg/cpp/src/barretenberg/trace_to_polynomials/trace_to_polynomials.cpp

```diff
@@ -38,28 +38,29 @@ void TraceToPolynomials<Flavor>::populate(Builder& builder,
         compute_permutation_argument_polynomials<Flavor>(builder, polynomials, copy_cycles, active_region_data);
     }
 }
-
 template <class Flavor>
 std::vector<CyclicPermutation> TraceToPolynomials<Flavor>::populate_wires_and_selectors_and_compute_copy_cycles(
     Builder& builder, ProverPolynomials& polynomials, ActiveRegionData& active_region_data)
 {
-
     PROFILE_THIS_NAME("construct_trace_data");
 
+    // Shift only for UltraZKFlavor
+    constexpr bool is_ultra_zk = std::is_same_v<Flavor, UltraZKFlavor>;
```

Comment:
again, with the right changes to the `UltraExecutionTraceBlocks` hopefully this file can be virtually unchanged